src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/WhenProcessingEnvelopeTest.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.controllers;
+
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import com.microsoft.azure.storage.core.PathUtility;
+import io.restassured.RestAssured;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.File;
+import java.net.URI;
+import java.util.concurrent.TimeUnit;
+
+import static com.jayway.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.is;
+
+@RunWith(SpringRunner.class)
+@TestPropertySource("classpath:application.properties")
+public class WhenProcessingEnvelopeTest {
+
+    @Value("${test-url}")
+    private transient String testUrl;
```

Comment:
Why are these things transient?

Reply:
The transient keyword in Java is used to indicate that a field should not be serialized. Why do we need it here that was the question and nothing else 😉

Reply:
It's probably one of the annotations making it considered a bean. An beans need to be serializable. We might want to start using jUnitRunner instead. Currently, our tests are starting a Spring app, which for funtional/smoke tests is completely unnecessary.

In some of our projects we have already started using https://github.com/lightbend/config to read configuration without Spring's help.